multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenAccessFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.machine.authentication.server;
+
+import static java.util.Arrays.asList;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import javax.ws.rs.Path;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.everrest.CheMethodInvokerFilter;
+import org.everrest.core.Filter;
+import org.everrest.core.resource.GenericResourceMethod;
+
+/**
+ * Limits set of methods which can be invoked using machine token signed requests.
+ *
+ * @author Max Shaposhnik (mshaposh@redhat.com)
+ */
+@Filter
+@Path("/{path:.*}")
+public class MachineTokenAccessFilter extends CheMethodInvokerFilter {
+
+  private final SetMultimap<String, String> allowedMethods = HashMultimap.create();
+
+  public MachineTokenAccessFilter() {
+    allowedMethods.putAll(
+        "/workspace", asList("getByKey", "addProject", "updateProject", "deleteProject"));
+    allowedMethods.putAll("/ssh", asList("getPair", "generatePair"));
+    allowedMethods.put("/preferences", "find");
+    allowedMethods.put("/activity", "active");
```

Comment:
There one more service which should be accessible with machine token - it's Factory Service. https://github.com/eclipse/che/issues/10710#issuecomment-420010260
Are there any plans to make it accessible?

Reply:
> getFactoryJson is already covered by permissions

It's true. But it won't be allowed since it is not listed in allowed methods here. Am I wrong?

Reply:
Oh, yes, you're correct